libc-top-half/musl/src/internal/atomic.h

```diff
@@ -182,13 +182,26 @@ static inline void a_dec(volatile int *p)
 #define a_store a_store
 static inline void a_store(volatile int *p, int v)
 {
+#ifdef __wasilibc_unmodified_upstream
```

Comment:
Would it be better so not defined `a_barrier` in `wasm32/atomic_arch.h` if its not doing the right thing?